platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java

```diff
@@ -851,14 +799,91 @@ public void deserialize(SerializableDataInputStream inStream, int version) throw
     }
 
+    /**
+     * Translates this address to the string format used in config.txt.
+     *
+     * @return the string form of this address that would appear in config.txt.
+     */
+    public String toConfigText() {
+        return "address, " + nickname + ", " + selfName + ", " + stake + ", " + ipString(addressInternalIpv4) + ", "
+                + portInternalIpv4 + ", " + ipString(addressExternalIpv4) + ", " + portExternalIpv4;
+    }
+
+    /**
+     * Parses an address from a single line of text.  The address must be in the form produced by
+     * {@link #toConfigText()}
+     *
+     * @param addressLine the string to parse an Address form.
+     * @param id the id to give to the parsed Address.
+     * @param isOwnHostDeterminer a function to determine if isOwn should be true given an InetAddress.
+     * @param memo the memo text for the address.
+     * @return the Address parsed from the addressLine.
+     * @throws ParseException if there is any problem with creating an Address from the addressLine.
+     */
+    public static Address parseConfigText(
+            String addressLine, long id, Function<InetAddress, Boolean> isOwnHostDeterminer, String memo)
+            throws ParseException {
+        String[] parts = addressLine.trim().split(",");
+        for (int i = 0; i < parts.length; i++) {
+            parts[i] = parts[i].trim();
+        }
+        if (!parts[0].equals("address")) {
+            throw new ParseException("The address line must start with 'address' and not '" + parts[0] + "'", 0);
+        }
+        final String nickname = parts[1];
+        final String selfname = parts[2];
+        Long stake;
+        try {
+            stake = Long.parseLong(parts[3]);
+        } catch (NumberFormatException e) {
+            throw new ParseException("Cannot parse value of stake from '" + parts[3] + "'", 3);
+        }
+        InetAddress internalIp;
+        try {
+            internalIp = InetAddress.getByName(parts[4]);
+        } catch (UnknownHostException e) {
+            throw new ParseException("Cannot parse ip address from '" + parts[4] + ",", 4);
+        }
+        int internalPort;
+        try {
+            internalPort = Integer.parseInt(parts[5]);
+        } catch (NumberFormatException e) {
+            throw new ParseException("Cannot parse ip port from '" + parts[5] + "'", 5);
+        }
+        InetAddress externalIp;
+        try {
+            externalIp = InetAddress.getByName(parts[6]);
+        } catch (UnknownHostException e) {
+            throw new ParseException("Cannot parse ip address from '" + parts[6] + ",", 6);
+        }
+        int externalPort;
+        try {
+            externalPort = Integer.parseInt(parts[7]);
+        } catch (NumberFormatException e) {
+            throw new ParseException("Cannot parse ip port from '" + parts[7] + "'", 7);
+        }
+        boolean isOwnHost = isOwnHostDeterminer.apply(internalIp);
```

Comment:
```suggestion
        final boolean isOwnHost = isOwnHostDeterminer.apply(internalIp);
```

Reply:
I added final keyword and moved function to AddressBookUtils.java